### DISCUSSION TITLE
How can I add htmlid to this special symbol, such as { or sqrt

### DISCUSSION THREAD

**Discussion by chenLe1232** [KaTeX/KaTeX#3217] [Q&A]
I try to perform some animation operations based on the HTML fragment rendered by Katex, but I find that I can't add htmlid to the following special characters.
```js
eg: \begin{Bmatrix}
   a & b \\
   c & d
\end{Bmatrix}
```
![image](https://user-images.githubusercontent.com/37372288/131483059-1491dd4c-fa43-477b-9196-52e4c32fa913.png)

**Comment by edemaine** [comment]
Indeed, it's difficult to add an `id` to a bracket made via `\left`.

Here's one approach to identifying the element:
```latex
\htmlId{my-matrix}{\begin{Bmatrix}
   a & b \\
   c & d
\end{Bmatrix}}
```
Then you can get the open bracket via `document.queryElement('#my-matrix .mopen')` or `document.queryElement('#my-matrix > .minner > .mopen`.

Another approach would be to avoid using `\left`, and instead use explicit sizing macros.  For example, this looks approximately the same:
```latex
\htmlId{my-matrix}{\bigg\{}\!\!
\begin{array}{cc}
   a & b \\
   c & d
\end{array}
\!\!\bigg\}
```

Hopefully one of these works for you!

**Reply by chenLe1232** [reply]
Please forgive me. I don't know how to express my gratitude. I can only choose to donate a little money on the home page every month, hoping to help you.
Thank you for answering this question. Of course, I'm willing to pay some consulting fees.

**Reply by edemaine** [reply]
Glad I could help! No need to pay me, but thank you for supporting KaTeX!

**Reply by chenLe1232** [reply]
I was surprised to see the second method. I met this writing method for the first time. How can I learn the knowledge of similar macros? Can you recommend relevant learning resources or website? Thank you very much.

**Reply by edemaine** [reply]
One good resource is KaTeX's own documentation of supported commands: https://katex.org/docs/supported.html#delimiters

Here's one general LaTeX resource which briefly covers delimiter sizing: https://www.overleaf.com/learn/latex/Brackets_and_Parentheses
